addons/membership/membership.py

```diff
@@ -488,6 +488,15 @@ def action_cancel(self, cr, uid, ids, context=None):
             member_line_obj.write(cr, uid, mlines, {'date_cancel': today})
         return super(Invoice, self).action_cancel(cr, uid, ids, context=context)
 
+    def unlink(self, cr, uid, ids, context=None):
+        member_line_obj = self.pool.get('membership.membership_line')
+        for invoice in self.browse(cr, uid, ids, context=context):
+            mlines = member_line_obj.search(cr, uid,
```

Comment:
What Odoo discourages are only cosmetic changes (like the spaces removal). But this is not cosmetic, it's a refactoring that avoids to change in case of a bug in two places, so I think it worths.